functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+
+.PARAMETER Scripts
+Switch to indicate if T-SQL restore scripts should be written out
+
+.PARAMETER ScriptOnly
+Switch indicates that ONLY T-SQL scripts should be generated, no restore takes place
+
+.PARAMETER VerifyOnly
+Switch indicate that restore should be verified
+
+.PARAMETER XpDirTree
+Switch that indicated file scanning should be performed by the SQL Server instance using xp_dirtree
+You must have sysadmin role membership on the instance for this to work.
+
+.PARAMETER LogicalFileMapping
+Accepts a hashtable of mappings to be used to rename Logical Files post restore, of the form:
+$mapping = @{'LogicalFile1'='newname1';'LogicalFile2'='othername'}
+You don't have to specify all files, and just those mapped will  be renamed
+Exclusive with LogicalFilePrefix
+
+.PARAMETER LogicalFilePrefix
+Specify a string which will be prefixed to ALL logical files post restore.
+Exlusive with LogicalFileMapping
+.NOTES
+Original Author: Stuart Moore (@napalmgram), stuart-moore.com
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.EXAMPLE
+Restore-DbaBackup -SqlServer server1\instance1 -path \\server2\backups\$ 
+
+Scans all the backup files in \\server2\backups$, filters them and restores the database to server1\instance1
+
+.EXAMPLE
+Restore-DbaBackup -SqlServer server1\instance1' -path \\server2\backups\$ -OlaStyle -RestoreLocation c:\restores
+
+Scans all the backup files in \\server2\backups$ stored in an Ola Hallengreen style folder structure,
+ filters them and restores the database to the c:\restores folder on server1\instance1 
+
+.EXAMPLE
+Get-ChildItem c:\SQLbackups1\, \\server\sqlbackups2 | Restore-DbaBackup -SqlServer server1\instance1 
+
+Takes the provided files from multiple directories and restores them on  server1\instance1 
+
+.EXAMPLE
+$RestoreTime = Get-Date('11:19 23/12/2016')
+Restore-DbaBackup -SqlServer server1\instance1' -path \\server2\backups\$ -OlaStyle -RestoreLocation c:\restores -RestoreTime $RestoreTime
+
+Scans all the backup files in \\server2\backups$ stored in an Ola Hallengreen style folder structure,
+ filters them and restores the database to the c:\restores folder on server1\instance1 up to 11:19 23/12/2016
+#>
+	[CmdletBinding()]
+	param (
+        [parameter(Mandatory = $true, ParameterSetName="Paths")]
+        [string[]]$Path,
```

Comment:
The set of functions is written on the basis that we don't have the origin server.

Would be easy enough to add a switch and another file getting script, and then carry on as normal.